crates/testing/src/task_helpers.rs

```diff
@@ -148,15 +235,67 @@ async fn build_quorum_proposal_and_signature(
         timestamp: 0,
         proposer_id: *api.public_key(),
     };
-    let signature = <BLSPubKey as SignatureKey>::sign(private_key, leaf.commit().as_ref())
+
+    let mut signature = <BLSPubKey as SignatureKey>::sign(private_key, leaf.commit().as_ref())
         .expect("Failed to sign leaf commitment!");
-    let proposal = QuorumProposal::<TestTypes> {
-        block_header,
-        view_number: ViewNumber::new(view),
+    let mut proposal = QuorumProposal::<TestTypes> {
+        block_header: block_header.clone(),
+        view_number: ViewNumber::new(1),
         justify_qc: QuorumCertificate::genesis(),
         timeout_certificate: None,
         proposer_id: leaf.proposer_id,
     };
+    for cur_view in 2..=view {
+        consensus.state_map.insert(
+            ViewNumber::new(1),
+            View {
+                view_inner: ViewInner::Leaf {
+                    leaf: leaf.commit(),
+                },
+            },
+        );
+        consensus.saved_leaves.insert(leaf.commit(), leaf.clone());
+        let quorum_membership = handle.hotshot.inner.memberships.quorum_membership.clone();
+        let quorum_data = QuorumData {
+            leaf_commit: leaf.commit(),
+        };
+        let created_qc = build_cert::<
+            TestTypes,
+            QuorumData<TestTypes>,
+            QuorumVote<TestTypes>,
+            QuorumCertificate<TestTypes>,
+        >(
+            quorum_data,
+            &quorum_membership,
+            ViewNumber::new(cur_view - 1),
+            public_key,
+            private_key,
+        );
+        let parent_leaf = leaf.clone();
+        let leaf_new_view = Leaf {
+            view_number: ViewNumber::new(cur_view),
+            justify_qc: created_qc.clone(),
+            parent_commitment: parent_leaf.commit(),
```

Comment:
Nit: I think we could just use `leaf.commit()` without creating `parent_leaf`. But maybe this whole construction isn't necessary if we could use the leaf from `saved_leaves` directly (see my other comment).

Reply:
Here I use `leaf` and `parent_leaf` separately, because for every loop it'll do:
```
let parent_leaf = leaf.clone();
let leaf_new_view = Leaf { ..., parent_leaf.commit(), ...};
leaf = leaf_new_view;
```

Maybe the structure is not the best, let me know if there's a better way to do the loop.

Reply:
I just think it's probably unnecessary to create `parent_leaf`. Maybe something like:

```rust
let leaf_new_view = Leaf { ..., leaf.clone().commit(), ...};
leaf = leaf_new_view;
```

Reply:
I just realized I should add more comments for code read-ability, hope the new comments could help QAQ.
And on the second thought of this, I'm more inclined to keep `let parent_leaf = leaf.clone();` if it will not introduce more latency or storage, as it serves as a change point of views, which could help with readability (I think). Would love to hear different sound if there's a better reason.

Reply:
Sounds good. I have no strong preference for this.